go/vt/schema/ddl_strategy.go

```diff
@@ -115,6 +116,14 @@ func ParseDDLStrategy(strategyVariable string) (*DDLStrategySetting, error) {
 	if _, err := setting.RetainArtifactsDuration(); err != nil {
 		return nil, err
 	}
+
+	switch setting.Strategy {
+	case DDLStrategyVitess, DDLStrategyOnline, DDLStrategyMySQL, DDLStrategyDirect:
+		if opts := setting.RuntimeOptions(); len(opts) > 0 {
+			return nil, fmt.Errorf("invalid flags for %v strategy: %s", setting.Strategy, strings.Join(opts, " "))
+		}
+	}
```

Comment:
I guess if it's some other strategy (or maybe there are none?) then there's no strategy flags?

Reply:
All strategies support flags such as `--postpone-launch`. A couple strategies (`gh-ost`, `pt-osc`) support "runtime options", which are flags passed to the underlying OSC binary. The specific flags used by these binaries are not listed nor managed by vitess. And so vitess accepts that some flags are unrecognized and it passed them blindly to the binary.

For `vitess`, `direct`, `mysql`, and because these are not "binaries", we reject any unknown flags.